DeviceAdapters/Dragonfly/ConfigFileHandler.cpp

```diff
@@ -0,0 +1,137 @@
+#include "ConfigFileHandler.h"
+#include <fstream>
+#include <windows.h>
+#include "Dragonfly.h"
+#include <Shlobj.h>
+
+using namespace std;
+
+const char* const g_SectionName = "config";
+const char* const g_DefaultConfigFileName = "DragonflyMMConfig.ini";
+const char* const g_ConfigFile = "Configuration file";
+
+#if defined(WIN32) || defined(_WIN32) 
+#define PATH_SEPARATOR "\\" 
+#else 
+#define PATH_SEPARATOR "/" 
+#endif 
+
+CConfigFileHandler::CConfigFileHandler( CDragonfly* MMDragonfly )
+  : MMDragonfly_( MMDragonfly )
+{
+  FileName_ = "Undefined";
+
+  // Create property with a default path set to My Documents
+  PWSTR vWPathName;
+  if ( SUCCEEDED( SHGetKnownFolderPath( FOLDERID_ProgramData, KF_FLAG_DEFAULT, NULL, &vWPathName ) ) ) // FOLDERID_PublicDocuments for Public documents path - FOLDERID_ProgramData for All users (aka ProgramData) path
+  {
+    wstring vWPathNameString = vWPathName;
+    FileName_.assign( vWPathNameString.begin(), vWPathNameString.end() );
```

Comment:
Here I get the warning

```text
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xstring(2541,23): warning C4244: 'argument': conversion from 'wchar_t' to 'const _Elem', possible loss of data
        with
        [
            _Elem=char
        ]
```

as a result of passing a pair of wchar_t iterators to std::string::assign (it will not autoconvert but rather take the low byte of each wide character).

I think this is safe if the string is guaranteed to be ASCII only. Is that a good assumption here (e.g. on non-English versions of Windows)?

Reply:
Good point. I now use ```wcstombs_s``` to do the conversion from ```wstring``` to ```string```. From what I gathered it may not be bulletproof but there's an easy workaround for the user if the conversion doesn't work properly.